game-app/game-core/src/main/java/games/strategy/triplea/attachments/TerritoryAttachment.java

```diff
@@ -264,14 +268,7 @@ private void resetIsImpassable() {
   }
 
   public void setCapital(final String value) throws GameParseException {
-    if (value == null) {
-      capital = null;
-      return;
-    }
-    final GamePlayer p = getData().getPlayerList().getPlayerId(value);
-    if (p == null) {
-      throw new GameParseException("No Player named: " + value + thisErrorMsg());
-    }
+    getPlayerOrThrow(value);
```

Comment:
Is this a no-op for null values? Otherwise we'd get an exception when passing null here, right?

Reply:
Null isn't supposed to be passed to setters on attachments. Some test were doing that, which are fixed.

Otherwise, the setters come from parsing XML attributes, which won't ever be null.